lib/api/tensor-impl.ts

```diff
@@ -97,17 +95,12 @@ export class Tensor implements TensorInterface {
       if (dim >= this.dims[idx]) {
         throw new RangeError(`Input index array dims don't match the tensor dims.`);
       }
-      flatIndices += idx < indexArray.length - 1 ? dim * this.dims.slice(idx + 1).reduce((a, b) => a * b) : dim;
     });
 
     if (typeof value === 'boolean') {
-      this.data[flatIndices] = value ? 1 : 0;
-    } else if (typeof value === 'string') {
-      this.data[flatIndices] = value;
-    } else if (ArrayBuffer.isView(this.data)) {
-      this.data.set([value], flatIndices);
```

Comment:
btw - do we need to couple API change with ndarray removal ?

Reply:
ndarray removal does not affect any API.